src/plugins/validation/2and3/semantic-validators/pagination-ibm.js

```diff
@@ -0,0 +1,181 @@
+module.exports.validate = function({ jsSpec }, config) {
+  const result = {};
+  result.error = [];
+  result.warning = [];
+
+  config = config.pagination;
+  const queryParameters = [];
```

Comment:
By defining and initializing `queryParameters` here, and not re-initializing it inside the loop, you are collecting up the query parameters for _all_ the operations in the spec into this list. That is definitely not right.

It would be better to move the definition and initialization of `queryParameters` inside the block that deals with a specific operation -- that will signal that it's value is intended to be just for that operation.